test/CMakeLists.txt

```diff
@@ -205,6 +205,10 @@ if(NRN_ENABLE_PYTHON AND PYTEST_FOUND)
     ENVIRONMENT COVERAGE_FILE=.coverage.cover_tests
     COMMAND ${PYTHON_EXECUTABLE} ${pytest} ./test/cover
     SCRIPT_PATTERNS test/cover/*.py test/cover/*.json)
+  nrn_add_test(
+    GROUP coverage_tests
+    NAME example_nmodl
+    COMMAND ${PROJECT_BINARY_DIR}/bin/nrnivmodl ${PROJECT_SOURCE_DIR}/share/examples/nrniv/nmodl)
```

Comment:
If you point `MODFILE_PATTERNS` to the relevant MOD files then `nrn_add_test` will take care of running `nrnivmodl`, and you can use `COMMAND` to run some command using the `special` binary that is produced.

Reply:
Did so. Sadly, though it forced me to create a new GROUP as, apparently one cannot override MODFILE_PATTERNS for a new test in that group. Also, it may be that it will fail cmake_format as I can't seem to make that work on my Mac with ```-DNRN_CMAKE_FORMAT=ON```

Reply:
Indeed, a group must use the same set of MOD files. There shouldn't be too much overhead with a new group, though.

For the formatting, https://github.com/neuronsimulator/nrn/pull/1982 was perhaps not very well advertised/documented. You should be able to do `./external/coding-conventions/bin/format any/file/to.format`, independently of any CMake options (or whether CMake has been run at all).

Reply:
we should we update the documentation on formatting, do some CMake clean-up

Reply:
will do another PR for the docs formatting